src/main/kotlin/no/nav/syfo/arbeidsgivere/service/ArbeidsgiverService.kt

```diff
@@ -0,0 +1,128 @@
+package no.nav.syfo.arbeidsgivere.service
+
+import io.ktor.util.KtorExperimentalAPI
+import no.nav.syfo.arbeidsgivere.client.arbeidsforhold.client.ArbeidsforholdClient
+import no.nav.syfo.arbeidsgivere.client.arbeidsforhold.model.Arbeidsavtale
+import no.nav.syfo.arbeidsgivere.client.arbeidsforhold.model.Arbeidsforhold
+import no.nav.syfo.arbeidsgivere.client.narmesteleder.NarmesteLederRelasjon
+import no.nav.syfo.arbeidsgivere.client.narmesteleder.NarmestelederClient
+import no.nav.syfo.arbeidsgivere.client.organisasjon.client.OrganisasjonsinfoClient
+import no.nav.syfo.arbeidsgivere.client.organisasjon.model.Navn
+import no.nav.syfo.arbeidsgivere.client.organisasjon.model.Organisasjonsinfo
+import no.nav.syfo.arbeidsgivere.model.Arbeidsgiverinfo
+import no.nav.syfo.arbeidsgivere.model.NarmesteLeder
+import no.nav.syfo.arbeidsgivere.redis.ArbeidsgiverRedisService
+import no.nav.syfo.arbeidsgivere.redis.toArbeidsgiverinfo
+import no.nav.syfo.arbeidsgivere.redis.toArbeidsgiverinfoRedisModel
+import no.nav.syfo.client.StsOidcClient
+import no.nav.syfo.log
+import no.nav.syfo.pdl.service.PdlPersonService
+import java.time.LocalDate
+import kotlin.random.Random
+
+@KtorExperimentalAPI
+class ArbeidsgiverService(
+    private val arbeidsforholdClient: ArbeidsforholdClient,
+    private val organisasjonsinfoClient: OrganisasjonsinfoClient,
+    private val narmestelederClient: NarmestelederClient,
+    private val pdlPersonService: PdlPersonService,
+    private val stsOidcClient: StsOidcClient,
+    private val arbeidsgiverRedisService: ArbeidsgiverRedisService
+) {
+    suspend fun getArbeidsgivere(fnr: String, token: String, date: LocalDate, sykmeldingId: String): List<Arbeidsgiverinfo> {
+        val arbeidsgivereFraRedis = getArbeidsgivereFromRedis(fnr)
+        if (arbeidsgivereFraRedis != null) {
+            log.info("Fant arbeidsgivere i redis")
+            return arbeidsgivereFraRedis
+        }
+
+        val stsToken = stsOidcClient.oidcToken()
+        val person = pdlPersonService.getPerson(fnr = fnr, userToken = token, callId = sykmeldingId, stsToken = stsToken.access_token)
+        val ansettelsesperiodeFom = LocalDate.now().minusMonths(4)
+        val arbeidsgivere = arbeidsforholdClient.getArbeidsforhold(fnr = fnr, ansettelsesperiodeFom = ansettelsesperiodeFom, token = token, stsToken = stsToken.access_token)
+        val aktiveNarmesteledere = narmestelederClient.getNarmesteledere(person.aktorId).filter { it.aktivTom == null }
+        val arbeidsgiverList = ArrayList<Arbeidsgiverinfo>()
+        arbeidsgivere.filter {
+            it.arbeidsgiver.type == "Organisasjon"
+        }.forEach { arbeidsforhold ->
+            val organisasjonsinfo =
+                organisasjonsinfoClient.getOrginfo(arbeidsforhold.arbeidsgiver.organisasjonsnummer!!)
```

Comment:
Om det blir mange oppslag mot diverse tjenester så kan man gjøre de async, men vi får se hvor lang tid det tar

Reply:
Skulle ønske vi hadde et graphql-api for organisasjon sånn at vi kunne hente alle orgnavnene på en gang :(